web/src/repo/settings/RepoSettingsCodeIntelligencePage.tsx

```diff
@@ -0,0 +1,153 @@
+
+interface Props extends RouteComponentProps<any> {
+    repo: GQL.IRepository
+}
+
+interface State {
+    latestDumps?: GQL.ILSIFDump[]
+    loading: boolean
+    error?: Error
+}
+
+/**
+ * The repository settings code intelligence page.
+ */
+export class RepoSettingsCodeIntelligencePage extends React.PureComponent<Props, State> {
+    public state: State = { loading: true }
+
+    private updates = new Subject<void>()
+    private subscriptions = new Subscription()
+
+    public componentDidMount(): void {
+        eventLogger.logViewEvent('RepoSettingsCodeIntelligence')
+
+        this.subscriptions.add(
+            this.updates
+                .pipe(
+                    tap(() => this.setState({ loading: true })),
+                    switchMap(() => this.queryLatestDumps())
+                )
+                .subscribe(
+                    c =>
+                        this.setState({
+                            latestDumps: c.nodes.sort((a, b) => a.projectRoot.path.localeCompare(b.projectRoot.path)),
+                            loading: false,
+                        }),
+                    error => this.setState({ error, loading: false })
+                )
+        )
+        this.updates.next()
+    }
+
+    public componentWillUnmount(): void {
+        this.subscriptions.unsubscribe()
+    }
+
+    public render(): JSX.Element | null {
+        return (
+            <div className="repo-settings-code-intelligence-page">
+                <PageTitle title="Code Intelligence" />
+                <h2>Code Intelligence</h2>
+                <p>
+                    Enable precise code intelligence by{' '}
+                    <a href="https://docs.sourcegraph.com/user/code_intelligence/lsif">uploading LSIF data</a>.
+                </p>
+
+                <div className="lsif-dump-collection">
+                    <h3>Current LSIF Uploads</h3>
+                    <p>
+                        These uploads provide code intelligence for the latest commit and are used in cross-repository{' '}
+                        <em>Find Reference</em> requests.
+                    </p>
+
+                    {this.state.loading && <LoadingSpinner className="icon-inline" />}
+                    {this.state.error && (
+                        <div className="alert alert-danger">
+                            Error getting repository index status:
+                            <br />
+                            <code>{this.state.error.message}</code>
+                        </div>
+                    )}
+                    {!this.state.error &&
+                    !this.state.loading &&
+                    this.state.latestDumps &&
+                    this.state.latestDumps.length > 0 ? (
+                        this.state.latestDumps.map((d, i) => <LsifDumpNode key={i} node={d} />)
+                    ) : (
+                        <p>No dumps are marked as latest for this repository.</p>
```

Comment:
```suggestion
                        <p>No dumps are recent enough to be used at the tip of the default branch.</p>
```

I tried to make this less about marking and more about recency, which users are more likely to understand.

Reply:
This is much better - thanks!